flask_app.py

```diff
@@ -53,11 +86,41 @@ def available_years(session: Session) -> list[int]:
 
 
 @app.route("/", methods=["GET"])
+@login_required
 def home():
     return redirect(url_for("report"))
 
 
+@app.route("/login", methods=["GET", "POST"])
+def login():
+    if hasattr(current_user, "is_authenticated") and current_user.is_authenticated:
```

Comment:
I read that flask-login guarantees `current_user` to always be an object with `is_authenticated` so we don't need `hasattr` here, just do:

```
    if current_user.is_authenticated:
        return redirect(url_for("report"))
```